connection.go

```diff
@@ -92,6 +93,10 @@ func (mc *mysqlConn) Close() (err error) {
 // closed the network connection.
 func (mc *mysqlConn) cleanup() {
 	// Makes cleanup idempotent
+	if mc.authPlugin != nil {
+		mc.authPlugin.Close()
```

Comment:
The auth plugin is not used again after the connection is established and can therefore be closed in the init phase already, I believe. We also don't have to add it as a field to ` mc` then.